src/main/java/com/openshift/internal/restclient/model/Container.java

```diff
@@ -142,13 +143,18 @@ public String getImagePullPolicy() {
 	}
 
 	@Override
-	public void setLifecycle(String lifecycle) {
-		set(node, propertyKeys, LIFECYCLE, lifecycle);
+	public void setLifecycle(ILifecycle lifecycle) {
+		ModelNode lifecycleNode = ModelNode.fromJSONString(lifecycle.toJson());
+		get(node, propertyKeys, LIFECYCLE).set(lifecycleNode);
 	}
 
 	@Override
-	public String getLifecycle() {
-		return asString(node, propertyKeys, LIFECYCLE);
+	public ILifecycle getLifecycle() {
+		if (node.has(LIFECYCLE)) {
+			return Lifecycle.fromJson(get(node, propertyKeys, LIFECYCLE));
+		} else {
```

Comment:
Ideally, I'd return Optional.empty, but that would require changing the API. Now it will return an 'empty' lifecycle, which is not ideal. Perhaps null is better.